projects/aws/bottlerocket-bootstrap/pkg/kubeadm/utils.go

```diff
@@ -251,3 +251,58 @@ func walkDirFunc(path string, dirEntry fs.DirEntry, err error) error {
 	}
 	return nil
 }
+
+// KubeletTlsConfig is a struct that holds cert and private key
+type KubeletTlsConfig struct {
+	KubeletServingCert       string
+	KubeletServingPrivateKey string
+}
+
+// readKubeletTlsConfig loads kubelet serving cert and private key from files
+func readKubeletTlsConfig(reader FileReader) *KubeletTlsConfig {
+	kubeletServingCertFile := "/.bottlerocket/rootfs/var/lib/kubeadm/pki/kubelet-serving.crt"
+	kubeletServingPrivateKeyFile := "/.bottlerocket/rootfs/var/lib/kubeadm/pki/kubelet-serving.key"
+
+	kubeletTlsConfig := &KubeletTlsConfig{}
+
+	kubeletServingCertContents, err := readAndEncodeFile(kubeletServingCertFile, reader)
+	if err != nil {
+		message := fmt.Errorf("skipping Kubelet TLS configuration: %s %w", kubeletServingCertFile, err)
+		fmt.Printf("%s\n", message)
+		return nil
+	}
+
+	kubeletServingPrivateKeyContents, err := readAndEncodeFile(kubeletServingPrivateKeyFile, reader)
+	if err != nil {
+		message := fmt.Errorf("skipping Kubelet TLS configuration: %s %w", kubeletServingPrivateKeyFile, err)
+		fmt.Printf("%s\n", message)
+		return nil
+	}
+	kubeletTlsConfig.KubeletServingCert = kubeletServingCertContents
+	kubeletTlsConfig.KubeletServingPrivateKey = kubeletServingPrivateKeyContents
```

Comment:
We should check if these are `!= ""` and then only set them else return nil, so that we dont always call the tls cert set api even on `""`

Reply:
Posted revision to address this.